source/base/mpi.cc

```diff
@@ -622,13 +622,49 @@ namespace Utilities
           return;
         }
 
+      /*
+       * Create a custom MPI datatype handle describing the memory layout
+       * of the MinMaxAvg struct and a custom MPI op handle for our
+       * max_reduce function.
+       *
+       * This setup will leak some minor allocated memory (in the range of
+       * a few bytes) at the end of the program run because we never call
+       * MPI_Type_free(&type) and MPI_Op_free(&op) to explicitly free up
+       * memory. This is a cosmetic issue but some address sanitizers will
+       * complain about this (such as clang's -fsanitize=address).
+       */
+      static MPI_Datatype type;
+      static MPI_Op       op;
+
+      std::once_flag initialization_flag;
```

Comment:
This isn't going to work unless you mark the variable as `static`. But I think there is a better option (see above).